spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -437,6 +448,27 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 			};
 		}
 
+		private void seekPartitions(Collection<TopicPartition> partitions, boolean idle) {
+			Map<TopicPartition, Long> current = new HashMap<>();
+			for (TopicPartition topicPartition : partitions) {
+				current.put(topicPartition,	ListenerConsumer.this.consumer.position(topicPartition));
+			}
+			ConsumerSeekCallback callback = new ConsumerSeekCallback() {
+
+				@Override
+				public void seek(String topic, int partition, long offset) {
+					ListenerConsumer.this.consumer.seek(new TopicPartition(topic, partition), offset);
```

Comment:
Plus, rebalance listener `onPartitionsAssigned()` is called from within the `poll()` so we need to seek before we actually poll for data. Otherwise we'd get data first and then seek.